File: app/src/main/java/finalproj/dressapp/httpclient/models/Product.java
package finalproj.dressapp.httpclient.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

class RentingDate {
    @SerializedName("fromdate")
    public String fromDate;

    @SerializedName("todate")
    public String toDate;
}

public class Product {
    @SerializedName("user")
    public String user;

    @SerializedName("id")
    public String id;

    @SerializedName("name")
    public String name;

    @SerializedName("price")
    public Number price;

    @SerializedName("fromdate")
    public String fromdate;

    @SerializedName("todate")
    public String todate;

    @SerializedName("image")
    public String image;

    @SerializedName("rentingDates")
    public List<RentingDate> rentingDates;

    public Product(String productName, Number price, String fromDate, String toDate, String image) {
        this.name = productName;
        this.price = price;
        this.fromdate = fromDate;
        this.todate = toDate;    
        this.image = image;
    }
}

File: server/src/models/Product.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

// Create Schema
const ProductSchema = new Schema({
    user: {
        type: Schema.Types.ObjectId,
        ref: "User",
        required: true
    },
    name: {
        type: String,
        required: true
    },
    price: {
        type: Number,
        required: true
    },
    fromdate: {
        type: Date,
        required: true
    },
    todate: {
        type: Date,
        required: true
    },
    image: {
        type: String,
        required: false
    },
    rentingDates: {
        type: [{
            fromdate: Date,
            todate: Date
        }],
        required: false
    }
});

ProductSchema.set('toJSON', {
    virtuals: true,
    versionKey: false,
    transform: function (doc, ret) { delete ret._id }
});

module.exports = Product = mongoose.model('Product', ProductSchema, 'Products');